src/Razor/src/Microsoft.VisualStudio.LanguageServices.Razor/LspEditorFeatureDetector.cs

```diff
@@ -56,22 +55,10 @@ public void Dispose()
     }
 
     private static async Task<bool> ComputeUseLegacyEditorAsync(
-        IVsService<SVsFeatureFlags, IVsFeatureFlags> vsFeatureFlagsService,
         IVsService<SVsSettingsPersistenceManager, ISettingsManager> vsSettingsManagerService,
         RazorActivityLog activityLog,
         CancellationToken cancellationToken)
     {
-        var vsFeatureFlags = await vsFeatureFlagsService.GetValueAsync(cancellationToken).ConfigureAwaitRunInline();
-
-        // IVsFeatureFlags is free-threaded but VSTHRD010 seems to be reported anyway.
-#pragma warning disable VSTHRD010 // Invoke single-threaded types on Main thread
-        if (vsFeatureFlags.IsFeatureEnabled(WellKnownFeatureFlagNames.UseLegacyRazorEditor, defaultValue: false))
```

Comment:
It's very surprising to me that this check was first.

Are there any active experiments running at the moment? Given that this check was first, it seems like its possible someone could have the "Use legacy editor" option turned on, but be given the LSP editor via this flag, and with this change we'll actually be sending them back to the legacy editor. Definitely don't want to create new legacy users :)

Reply:
I don't believe so, here's the one that flipped everyone to new Razor LSP, and it's currently stopped. https://exp.microsoft.com/a/feature/185937ef-6128-4980-b2bf-eb826ce7a056?workspaceId=d62eaecc-82a1-4029-a2ed-df68c376d916

Reply:
I guess if it was still running, the publicly available tools/options would've been broken for those users, which seems wrong, even if we don't want them to be on legacy :)